### ISSUE TITLE
Don't triggerDeprecationError

### ISSUE THREAD

**Issue by Avoran** [matthiasnoback/symfony-service-definition-validator#27]
As seen on https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/DependencyInjection/Definition.php, line 103 - 117:

```php
/**
     * Gets the factory class.
     *
     * @return string|null The factory class name
     *
     * @deprecated since version 2.6, to be removed in 3.0.
     */
    public function getFactoryClass($triggerDeprecationError = true)
    {
        if ($triggerDeprecationError) {
            @trigger_error('The '.__METHOD__.' method is deprecated since version 2.6 and will be removed in 3.0.', E_USER_DEPRECATED);
        }
        return $this->factoryClass;
    }
```
The `$triggerDeprecationError` boolean is introduced in most of the deprecated methods. I would like to see the calls you make to such classes pass false to disable the trigger_error. Not because I hate errors (actually, I do), but because we don't need this message anymore since you have your code 3.0 compatible.

**Comment by matthiasnoback** [comment]
I just fixed this and released the fix as v1.2.4. Thanks for pointing this out. If anything else triggers deprecation warnings, please let me know.
